core/src/main/java/org/apache/iceberg/GenericDataFile.java

```diff
@@ -174,12 +174,20 @@ private GenericDataFile(GenericDataFile toCopy) {
     this.fileSizeInBytes = toCopy.fileSizeInBytes;
     this.fileOrdinal = toCopy.fileOrdinal;
     this.sortColumns = copy(toCopy.sortColumns);
-    // TODO: support lazy conversion to/from map
-    this.columnSizes = copy(toCopy.columnSizes);
-    this.valueCounts = copy(toCopy.valueCounts);
-    this.nullValueCounts = copy(toCopy.nullValueCounts);
-    this.lowerBounds = SerializableByteBufferMap.wrap(copy(toCopy.lowerBounds));
-    this.upperBounds = SerializableByteBufferMap.wrap(copy(toCopy.upperBounds));
+    if (fullCopy) {
+      // TODO: support lazy conversion to/from map
+      this.columnSizes = copy(toCopy.columnSizes);
```

Comment:
columnSizes was not part of the STATS_COLUMN field in FilteredManifest class

Reply:
That's because it isn't required to run the filters in `FilteredManifest`. Those columns are the ones that need to be projected for the stats filter to work, but may not have been requested by the caller.